blacktie/jatmibroker-xatmi/src/main/java/org/jboss/narayana/blacktie/jatmibroker/core/transport/hybrid/stomp/StompManagement.java

```diff
@@ -67,28 +67,30 @@ public static Socket connect(String host, int port, String username, String pass
     }
 
     public static void send(Message message, OutputStream outputStream) throws IOException {
-        log.trace("Writing on: " + outputStream);
-        StringBuffer toSend = new StringBuffer(message.getCommand().toString());
-        toSend.append("\n");
-
-        Iterator<String> keys = message.getHeaders().keySet().iterator();
-        while (keys.hasNext()) {
-            String key = keys.next();
-            String value = message.getHeaders().get(key);
-            toSend.append(key);
-            toSend.append(":");
-            toSend.append(value);
+        synchronized (outputStream) {
```

Comment:
Why can't you use

    java.net.socket.getChannel().write()

which probably requires fewer changes than you are making in the current PR.